home/models.py

```diff
@@ -839,6 +841,34 @@ def submit_whatsapp_template(self, previous_revision):
 
         return self.whatsapp_template_name
 
+    def get_all_links(self):
+        page_links = []
+        orderedcontentset_links = []
+
+        usage = ReferenceIndex.get_references_to(self).group_by_source_object()
+        for ref in usage:
+            for link in ref[1]:
+                if link.model_name == "content page":
+                    link_type = "Related Page"
+                    tab = "#tab-promotional"
+                    if link.related_field.name == "whatsapp_body":
+                        link_type = "WhatsApp: Go to button"
+                        tab = "#tab-whatsapp"
+
+                    page = ContentPage.objects.get(id=link.object_id)
+                    url = reverse("wagtailadmin_pages:edit", args=(link.object_id,))
+                    page_links.append((url + tab, f"{page} - {link_type}"))
+
+                elif link.model_name == "Ordered Content Set":
+                    orderedcontentset = OrderedContentSet.objects.get(id=link.object_id)
+                    url = reverse(
+                        "wagtailsnippets_home_orderedcontentset:edit",
+                        args=(link.object_id,),
+                    )
+                    orderedcontentset_links.append((url, orderedcontentset.name))
```

Comment:
Should we have an `else` here that logs an error? So that we're at least aware if in the future someone adds an additional reference that's not covered by these two?